projects/output-and-error-handling/main.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+func main() {
+	c := http.Client{Timeout: time.Duration(1) * time.Second}
+
+	resp, err := c.Get("http://localhost:8080")
+
+	if err != nil {
+		fmt.Fprint(os.Stderr, "Something went wrong. Cannot complete request at the moment\n")
+		return
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case 429:
+		retryTime := resp.Header.Get("Retry-After")
+		formattedRetryTime, err := parseRetryValue(retryTime)
+
+		if err != nil {
+			// cannot determine how long to sleep for, retry once after 1 sec delay. Reason - The server response shows it is safe to retry. However we might not want keep at it for long - hence the choice
+
+			sleep(1)
+			resp, err = retry(1, c)
+
+			if err != nil {
+				// retry still unsuccessful and server does not respond
+				fmt.Fprint(os.Stderr, "Something went wrong. Cannot complete request at the moment\n")
+				resp.Body.Close()
```

Comment:
You don't need this explicit call because you already called `defer resp.Body.Close()` which will make sure `resp.Body.Close()` gets called when the function returns. (Same below)